### PR TITLE
update server endpoints

### PR DESCRIPTION
**Pull request by hilvitzs**
Add GET endpoints for all mentors, students, and programs.
Add GET endpoints for individual mentor, student and prgoram.
Add POST for mentors, students and programs.
Add PATCH for mentor, student, and program.
Add DELETE for mentor, student, and program.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+      } else {
+        response.status(201).json({ error: 'You are missing a property'});
+      }
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.post('/api/v1/programs', (request, response) => {
```
Comment (letakeane) [inline server.js]: I think we can leave this one out. It'd be easy enough for future people to add, if Turing adds programs.

File: server.js [hunk 2]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+  database('students').where('id', id).update(update)
+    .then(update => {
+      response.status(200).json({ update });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.patch('/api/v1/programs/:id', (request, response) => {
```
Comment (letakeane) [inline server.js]: Again, I don't think we need this.

File: server.js [hunk 3]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+app.get('/api/v1/programs/:id', (request, response) => {
+  const { id } = request.params;
+
+  database('programs').select()
+    .then(program => {
+      if (program) {
+        response.status(200).json(program)
+      }
+      return response.status(404).json({
+        error: 'There were no program found!'
```
Comment (JohnBinning) [inline server.js]: Should this be we could not find that program?

File: server.js [hunk 4]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+  database('programs').where('id', id).update(update)
+    .then(update => {
+      response.status(200).json({ update });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.delete('/api/v1/mentors/:id', (request, response) => {
```
Comment (letakeane) [inline server.js:206]: Do we want to secure this? Maybe tie that in with admin access only? So only Allison can delete mentors by request?

File: server.js [hunk 5]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+  database('mentors').where('id', id).del()
+    .then(qty => {
+      response.status(204).json({ qty });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.delete('/api/v1/students/:id', (request, response) => {
```
Comment (letakeane) [inline server.js:218]: See the above comment about securing this endpoint

File: server.js [hunk 6]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+  database('students').where('id', id).del()
+    .then(qty => {
+      response.status(204).json({ qty });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
+app.delete('/api/v1/programs/:id', (request, response) => {
```
Comment (letakeane) [inline server.js]: Don't think we need this endpoint.

File: server.js [hunk 7]
```diff
@@ -47,6 +47,228 @@ app.get('/authenticate', (request, response) => {
...
+
+  database('programs').where('id', id).del()
+    .then(qty => {
+      response.status(204).json({ qty });
+    })
+    .catch(error => {
+      response.status(500).json({ error });
+    });
+});
+
```
Comment (letakeane) [inline server.js]: Thanks for writing these, Spencer!

**Review by letakeane — COMMENTED** [review]
Thank you!

File: server.js (hunk 4 above)
Comment (JohnBinning) [reply server.js:206]: I think securing for admin access is a good idea.  I'll create an issue to revisit once we have established admin login capability.

**Pull request merged**